solr/core/src/java/org/apache/solr/update/UpdateLog.java

```diff
@@ -627,47 +626,89 @@ protected void initTlogDir(SolrCore core) {
     }
   }
 
-  // TODO SOLR-17458: Migrate to Otel
   @Override
   public void initializeMetrics(
       SolrMetricsContext parentContext, Attributes attributes, String scope) {
     solrMetricsContext = parentContext.getChildContext(this);
-    bufferedOpsGauge =
-        () -> {
-          if (state == State.BUFFERING) {
-            if (bufferTlog == null) return 0;
-            // numRecords counts header as a record
```

Comment:
there are comments in this confusing code that you didn't retain when you extracted the method.  Please do.